pkg/kubelet/kubelet.go

```diff
@@ -1063,7 +1063,13 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
 		return err
 	}
 
-	err = kl.containerRuntime.SyncPod(pod, runningPod, podStatus)
+	pullSecrets, err := kl.getPullSecretsForPod(pod)
```

Comment:
Add a TODO here that duplicate secrets are being pulled and that we probably need a secret manager interface in the future.